resource/config_map.go

```diff
@@ -0,0 +1,198 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ConfigMapConfig struct {
+	// DesiredFn is function returning a desired ConfigMap object for the
+	// given custom resource object. Returned ConfigMap must have name and
+	// namespace equal to the values of Name and Namespace fields.
+	DesiredFn func(context.Context, interface{}) (*corev1.ConfigMap, error)
```

Comment:
I thought if `DesiredFn` should return `[]*corev1.ConfigMap` instead. But this would make it weird with having `Name` and `Namespace`. And if creating separate resource for each object is too much overhead we can just create a resource `ConfigMapSet` using this one under the hood.

Reply:
I don't follow the issues with the name and namespace. We could say the object metadata name and namespace have to be given and then we use these for the creation and updates.